### ISSUE TITLE
0.15 release

### ISSUE THREAD

**Issue by marqh** [SciTools/cartopy#838]
I'd like to cut a release to get out a few new features and fixes

this is a note to maintainers that this is on the way, and a milestone to add to your PRs if you'd like me to try and include them in the release

cheers
mark

**Comment by QuLogic** [comment]
I added a couple of mine that are bug fixes. What about things that are targetting 0.14? Should I just bump those?

**Comment by malcolmbrooks** [comment]
Hi Mark,
after #815 the build is failing with licence errors.
I think that changing the change to test_images.py was all done in 2016 in my branch (so passed the tests), but relative to upstream/master the change was in 2017 so the tests fail. I think fixing the copyright date line at the top will fix the build.

Malcolm

**Comment by ghisvail** [comment]
#799 and #804 would be nice ones to fix for the Debian packaging.

**Comment by marqh** [comment]
> I added a couple of mine that are bug fixes. What about things that are targetting 0.14? Should I just bump those?

Hi @QuLogic 

i'm sorry, i missed these questions, or missed answering them

in my view, anything that is still targeting 0.14 needs to stop doing so, we should close that milestone.

which 'bug fix' PRs are outstanding and ready to go from your point of view please?

mark

**Comment by QuLogic** [comment]
* #782 is a clear bug fix that is ready to go.
* #827 fixes compatibility with new OWSLib.

**Comment by lbdreyer** [comment]
Two outstanding issues that need to be addressed before the release:
* Cartopy still needs a whats new before we can do the release
* There seems to be an issue with a couple of the images in the gallery (e.g. see https://lbdreyer.github.io/scitools.org.uk/cartopy/docs/latest/examples/effects_of_the_ellipse.html)

**Comment by dopplershift** [comment]
Is everything working with matplotlib 2.0?

**Comment by QuLogic** [comment]
That is #773, so unfortunately not. I'd be happy to rebase later today if it's a priority but a decision must be made on what to do with images that changed.

**Comment by dopplershift** [comment]
IMO, it seems silly to cut a new release that's not 2.0-compatible. But I'm not plugged in enough to know what's driving the 0.15 release.

**Comment by bjlittle** [comment]
@dopplershift I think that releases are cheap, and personally I'd be keen to see a little more momentum on cartopy, starting with regular small releases. 

I don't think that mpl v2.0 support should be a blocker to a v0.15 release, any thoughts @marqh and @ajdawson ?

**Comment by marqh** [comment]
> IMO, it seems silly to cut a new release that's not 2.0-compatible. But I'm not plugged in enough to know what's driving the 0.15 release.

Hello @dopplershift 

I have had a look at cartopy's tests with respect to matplotlib 2.0m I did this a few days ago:
https://github.com/SciTools/cartopy/pull/843
as illustrated by
https://travis-ci.org/SciTools/cartopy/builds/196897760
there are a number of test failures under matplotlib 2.0, triggered by
```
import matplotlib.tests as mtests
```
I don't know how far back this hole goes and we are keen to get a release out as there are a couple of new features we want to provide to our user community right now.
I think it may take a bit of dedicated developer effort to get this all fixed up and we were not able to put this effort into the matplotlib release candidates so that we were ready for this eventuality.

We will come back round the loop and get this fixed, I have made a dedicated issue #848 to handle this and keep it at the forefront of our minds.  The next release really should have this properly sorted out.

On this basis, I think there is sufficient need for Cartopy 0.15 to release it without matplotlib 2.0 compatibility.  This is not ideal, but I think it is the pragmatic solution given our various constraints.

I agree with @bjlittle that we should proceed and cut this release.

thank you for highlighting this and encouraging us to do this, I'm afraid we're going to take the 'silly' option.  I am hanging my head in shame about this, but doing it anyway

**Comment by marqh** [comment]
> #782 is a clear bug fix that is ready to go.
> #827 fixes compatibility with new OWSLib.

@QuLogic both of these are merged now

**Comment by dopplershift** [comment]
Fair enough, and I certainly have nothing against more frequent releases.

**Comment by ghisvail** [comment]
Debian currently has `cartopy-0.14.2` with `matplotlib-2.0` and `owslib-0.14`. Sounds to me like cartopy won't be operating properly with this combination, am I correct?

**Comment by marqh** [comment]
> Debian currently has cartopy-0.14.2 with matplotlib-2.0 and owslib-0.14. Sounds to me like cartopy won't be operating properly with this combination, am I correct?

Hi @ghisvail 

I'm afraid that is the case, yes.  cartopy mostly works with matplotlib-2 but a number of tests are failing so I would not say it is an approved combination.

we need to fix this, see #848 but it will not be done until the next release

**Comment by ghisvail** [comment]
> I would not say it is an approved combination.

In this case, since we have yet to have any package depending on `cartopy` in the archive, it is safer to request its removal for the next stable version of Debian (for which we can no longer accept new upstream releases due to the freeze).

Meanwhile, I'll monitor progress on #848 and #797 which should help us achieve a good combination of packages in the Debian archive.

**Comment by marqh** [comment]
> In this case, since we have yet to have any package depending on cartopy in the archive, it is safer to request its removal for the next stable version of Debian (for which we can no longer accept new upstream releases due to the freeze).

that is pretty disappointing, from my point of view, but I do understand. 

We will get in touch as soon as matplotlib 2 compatibility is confirmed in a release and look to get this rectified

thank you for all your input on this @ghisvail

**Comment by ghisvail** [comment]
Now that version 0.15 (and 0.15.1) has been tagged, perhaps this bug can be closed? Also, would you guys consider pushing a release to PyPI. The current version available for download is still stuck at 0.14.2.

Thanks.

**Comment by gauteh** [comment]
If possible, please push a version to pypi.

**Comment by dopplershift** [comment]
@pelson 0.14.2 is still the latest version of CartoPy on PyPI--can you rectify that?

**Comment by ghisvail** [comment]
Any news of an upload of 0.15.x to PyPI?

**Comment by QuLogic** [comment]
@pelson I think this is something you need to handle.

**Comment by pelson** [comment]
@QuLogic - sorry for being a bit slow on this one. I'd be more than happy to add you to the appropriate role on pypi.python.org if that is helpful. I just need your pypi ID.

Ping all @SciTools/cartopy-devs for similar info.

**Comment by QuLogic** [comment]
You can use my username here for it.

**Comment by pelson** [comment]
Have done, thanks @QuLogic. I've also uploaded v0.15.1 with ```twine upload dist/Cartopy-0.15.1.tar.gz```. Closing this issue out. Please open another issue if there is a problem with the distribution on pypi.

**Comment by ghisvail** [comment]
Sadly no `docs` in the release tarball. Will open  a separate issue.

**Comment by ghisvail** [comment]
See #913

**Comment by pelson** [comment]
Thanks @ghisvail.
